src/webui/components/Login/index.js

```diff
@@ -32,20 +51,30 @@ export default class LoginModal extends Component {
    * set login modal's username and password to current state
    * Required by: <LoginModal />
    */
-  setCredentials(name, e) {
+  setCredentials(key, value) {
     this.setState({
-      [name]: e
+      form: Object.assign({}, this.state.form, {[key]: value})
```

Comment:
@ayusharma  why don't you use spread ...? I believe it's clearer.

Reply:
noted 🗒